0-intro-to-programming/slides-code/0-intro-to-programming.Rmd

```diff
@@ -0,0 +1,594 @@
+test run your code and find out where it breaks
+]
+
+.pull-right[
+  PICTURE TEXT ERROR IN HEART
+]
+
+
+---
+class: middle
+
+## When the code works you are only half-done
+
+How to make correct code into good code?
+
+Ask yourself 
+"_Would someone with your level of Python knowledge
+be able to follow the code?_"
+
+* Simplify
+* Peer-review
+* Document
+
+Settling for "_just_" correct code does not 
+save you or your team any time in the long run.
+
+
+---
+
+class: middle
+
+## Re-use code
+
+Since the instructions (code) is separate from input and output,
+we can re-use code blocks within and across projects. An imporant paradigm in coding is **DRY** (don't uepeat yourself)
+
+Three levels of re-using code:
+
+* **Copy-paste:** Copy a block a code of modify it to fit the context
+
+* **Functions:** A block of code you written that can be referenced across the project
+
+* **Library:** Code packages developed by others to solve difficult tasks but made easy use in any project
+
+***
+
+**Do not re-invent wheels**
+
+* Aim to have each section of code be built on someone else' efforts
+* Someone else's efforts can be the efforts of past you 
+* Compare with Excel - how do you build on the efforts of other users?
+
+---
+class: middle
+
+## Learn good code from expert coders
+
+.pull-left[
+  **Be careful** - expert code examples can be deceptive
+  
+  Advanced code examples is usually meant to explain
+  how to do something complex _correctly_, 
+  and not how to write _good code_.
+  
+  An advance example might not fit your project, 
+  regardless how _correct_ it solved a well-defined school book example.
+  Use critical thinking.
+]
+
+---
+class: middle
+
+## Learn good code from peer coders
+.pull-left[
+  Read code from real world data work examples, 
+  written by people at or just above your skill level
+  
+  This will teach you lessons 
+  much more applicable to your day-to-day work,
+  than school book examples
+  
+  Code written by someone at your level 
+  will teach you much more about how you
+  can become better at writing good code.
+  
+]
+
+---
+class: middle,
+
+## Documentation - comments
+
+.pull-left[
+
+  All programming languages has a feature called _comments_. 
+  
+  Comments are text in the same file as the code 
+  that the computer skips when running the code.
+  
+  Comments' only (but important) purpose
+  is to provide context for a human reading the code.
+  
+  Comments best compliment the code if 
+  they explains why the code is written a certain way.
+]
+
+.pull-right[
+
+  | OK comment | Good comment |
+  | --- | --- |
+  | _What_ code does | _What_ code does **and** _why_ |
+  | `Load data` | `Loading region definitions shared by NGO used in sampling` |
+  | `Deleting variable "id" and "name"` | `Removing identifying variables to create a non-confidential data set` |
+  
+]
+
+---
+class: inverse, center, middle
+
+# Some common concepts in programming 
```

Comment:
for this section I suggest adding a slide explaining what a script is and moving version control after code flow